h2o-k8s/src/main/java/water/k8s/lookup/KubernetesDnsLookup.java

```diff
@@ -0,0 +1,159 @@
+package water.k8s.lookup;
+
+
+import water.util.Log;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+import java.util.regex.Pattern;
+
+/**
+ * Discovery strategy on a DNS cluster leveraging the DNS record of a Kubernetes headless service present in the cluster.
+ * Kubernetes headless services, instead of load-balancing the requests onto one of the underlying pods, return the
+ * addresses of all the pods covered by the headless service.
+ * <p>
+ * Such pods can then be discovered via the above-mentioned DNS record. In order for H2O to know which service to query,
+ * it is mandatory for the K8S user to pass the name of the headless service to the H2O container, as follows:
+ *
+ * <pre>
+ * apiVersion: apps/v1
```

Comment:
nit: this is different from the readme. Specifically, open ports in container and also naming. Not sure if it matters, but this would maybe benefit from consistency.